types/relay-compiler/lib/GraphQLConsoleReporter.d.ts

```diff
@@ -0,0 +1,6 @@
+import { GraphQLReporter } from './GraphQLReporter';
+export default class GraphQLMultiReporter implements GraphQLReporter {
```

Comment:
This isn't a default export in the original library. It will still work since you consume it correctly in index.d.ts but it might be easier to maintain if it matched the js structure.

Reply:
I am not 100% sure what you meant, please check 07650b2. I hope this is what you requested.

Reply:
No, I meant these changes:

Named export (no default keyword):
`export class GraphQLMultiReporter ...`

And corresponding named import:
`import { MultiReporter } from './lib/GraphQLMultiReporter';`

Reply:
Are you 100% sure? I checked the code and it is definitely exported as `module.exports = GraphQLMultiReporter`.

Please also note that the folder structure in the git repository differs from the published npm package.

Reply:
`node_modules/relay-compiler/lib/GraphQLMultiReporter.js`

```js
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * 
 * @format
 */
'use strict';

var GraphQLMultiReporter =
/*#__PURE__*/
function () {
  function GraphQLMultiReporter() {
    for (var _len = arguments.length, reporters = new Array(_len), _key = 0; _key < _len; _key++) {
      reporters[_key] = arguments[_key];
    }

    this._reporters = reporters;
  }

  var _proto = GraphQLMultiReporter.prototype;

  _proto.reportMessage = function reportMessage(message) {
    this._reporters.forEach(function (reporter) {
      reporter.reportMessage(message);
    });
  };

  _proto.reportTime = function reportTime(name, ms) {
    this._reporters.forEach(function (reporter) {
      reporter.reportTime(name, ms);
    });
  };

  _proto.reportError = function reportError(caughtLocation, error) {
    this._reporters.forEach(function (reporter) {
      reporter.reportError(caughtLocation, error);
    });
  };

  return GraphQLMultiReporter;
}();

module.exports = GraphQLMultiReporter;
```

Reply:
Good catch on the github and npm code differing. But this actually means that you need to use the `module.exports = ` export in your type definitions as explained [here](https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-function-d-ts.html). Note that default export is different from this. All a `default export` does is add a property named `default` on your module and puts in the exported value into that property. Instead here you want the module itself to be a top level function/class.

Reply:
@PranavSenthilnathan Before changing everything:

Is the following correct?

```tsx
import { GraphQLSchema } from 'graphql';
export function transformASTSchema(
  schema: GraphQLSchema,
  transforms: string[],
): GraphQLSchema;
```

becomes

```tsx
import { GraphQLSchema } from 'graphql';
declare function transformASTSchema(schema: GraphQLSchema, transforms: string[]): GraphQLSchema;

declare const ASTConvert: { transformASTSchema: typeof transformASTSchema };

export = ASTConvert;
```

Reply:
`ASTConvert.d.ts` is fine as you already have it. However, `GraphQLMultiReporter.d.ts` needs to change to the `export =` syntax. You can read about the different library structures in the documentation [here](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html). The short answer is that if you exporting a top level class or function in a module, then you need to do `export =`. If you are not, and are just exporting an object with multiple useful members on it, then you can use the `export <foo>` listing syntax.

Reply:
Okay, now I get it! Thank you so much for the clarification.